### ISSUE TITLE
Modules symlinks not deleted (after reboot)

### ISSUE THREAD

**Issue by ckotte** [wesbarnett/snap-pac#21]
I have two symlinks that point to the modules directory of two previous kernels for whatever reason. This code should have deleted at least the 4.7.12 link during upgrade from 4.7.13 to 4.7.14.
```
find /usr/lib/modules -xtype l -delete
ln -sv /.snapshots/$(<$prefile)/snapshot/usr/lib/modules/$(uname -r) /usr/lib/modules/
```

Anways. I always get "errors" like this after booting the new kernel because the symlink to the old modules directory is still there:
```
[...]
( 1/18) Install DKMS modules
==> dkms install vboxhost/5.2.18_OSE -k 4.17.12-arch1-1-ARCH
mv: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxdrv.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxdrv.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxnetflt.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxnetflt.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxnetadp.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxnetadp.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxpci.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.12-arch1-1-ARCH/kernel/misc/vboxpci.ko': Read-only file system
==> dkms install vboxhost/5.2.18_OSE -k 4.17.14-arch1-1-ARCH
==> dkms install vboxhost/5.2.18_OSE -k 4.17.13-arch1-1-ARCH
mv: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxdrv.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxdrv.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxnetflt.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxnetflt.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxnetadp.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxnetadp.ko': Read-only file system
mv: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxpci.ko': Read-only file system
cp: cannot remove '/usr/lib/modules/4.17.13-arch1-1-ARCH/kernel/misc/vboxpci.ko': Read-only file system
( 2/18) Updating linux initcpios...
[...]
==> Building image from preset: /etc/mkinitcpio.d/linux.preset: 'fallback'
  -> -k /boot/vmlinuz-linux -c /etc/mkinitcpio.conf -g /boot/initramfs-linux-fallback.img -S autodetect
==> Starting build: 4.17.14-arch1-1-ARCH
  -> Running build hook: [base]
  -> Running build hook: [udev]
 [...]
```

How about adding a systemd service which deletes all modules symlinks during reboot? Just ```find /usr/lib/modules -xtype l -delete```!? 

The service could be activated (```systemctl enable```) or deactivated (```systemctl disable```) during snap-pac execution based on the LINK_MODULES configuration. If I enable LINK_MODULES, the service gets enabled automatically and if I disable it, it gets disabled.

I would like to implement it, but I want to get your feedback before. I can also just create my own service, but I would like to share and add it to this repo.

**Comment by wesbarnett** [comment]
There should be only one old symlink at most. I think we can just have pacman delete the old symlinks on the pre snapshot, since it should be the first thing run. Before anything is installed there. I have created a fix I'm about to push. Try it out and see if it works for you.

**Comment by wesbarnett** [comment]
No, this patch doesn't work correctly. I'm wondering if I should just take the logic of symlinking kernels out of the script since its causing all of these issues. It could probably be put into some other script and added as a pacman hook.

**Comment by wesbarnett** [comment]
pacman seems to check if conflicting files exist before any hooks are run, so the solution I committed won't work. A systemd unit could work, but again, it keeps adding complexity for a bash script that was supposed to just do pre and post snapshots

**Comment by wesbarnett** [comment]
This hook symlinks the kernel modules from the last pre snapshot.
 
    [Trigger]
    Operation = Upgrade
    Operation = Install
    Operation = Remove
    Type = Package
    Target = linux

    [Action]
    Description = Symlinking old kernel modules...
    When = PostTransaction
    Exec = /usr/bin/bash -c "ln -sv /.snapshots/$(snapper -c root list | awk '/pre/ { print $3 }' | tail -n 1)/snapshot/usr/lib/modules/$(uname -r) /usr/lib/modules/"

**Comment by wesbarnett** [comment]
Closed via #22.
